### ISSUE TITLE
Delta from tobiasjakobi's linux

### ISSUE THREAD

**Issue by nijhawank** [hexdump0815/linux-mainline-and-mali-on-odroid-u3#1]
It seems you have not taken everything added by Tobias Jakobi for enabling Mali and it's many components and may be other thing. Can you please specify what are the changes from his tree?

He also has a repository for libDRM with his changes. Isn't that needed?

**Comment by hexdump0815** [comment]
@nijhawank - i think i got all the mali relevant parts - if you go through his commit history they are clearly marked. in the end i picked the drivers/gpu/arm tree, Makefile adjustments and some other patches around (see the patches in the misc.exy dir) - for me at least it works (in the end even better than his original tree). for ubuntu 18.04 and debian buster the default libdrm seems to be fine meanwhile. please let me know in case you spot anything still missing.

btw. i'll provide bootable ubuntu 18.04 and debian 10 sd card images based on this tree here within the next days

best wishes - hexdump

**Comment by nijhawank** [comment]
Actually I'm not a developer in this space, just for my curiosity trying to understand a bit. I just knew that you have a CPU and a GPU before I realized that in the ARM world everything is separate (GPU, 2D, MFC etc.), So when I said earlier that you left some bits, I was referring to G2D /MFC which I understand Tobias is working on enabling in his tree (at least that's my understanding)
I now understand that Mali is only the G3D that you have already taken.

I agree that this tree is better than tobias's one as my odroid always got frozen within minutes of booting with his tree. It never happened with your patches. I tried your patches with 5.2.15 and it works amazing. Need to however enable the desktop (just have command line for time being)
